torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -60,88 +62,235 @@ def reset_table(self) -> None:
+            isinstance(token, lark.lexer.Token) and token.type == "CONSTANT"
         ):
-            if node.func.attr == "store":
-                # Do not allow for store to appear inside a read
-                # tl.load(a if tl.store(b) else z) is not useful
-                # and allowing this would complicate the analysis
-                assert self.read_depth == 0
-                assert self.in_store is False
-                self.in_store = True
-                self.generic_visit(node)
-                self.in_store = False
-                return
-            if node.func.attr in self.ALLOWED_READ_FNS:
-                self.read_depth += 1
-                self.generic_visit(node)
-                self.read_depth -= 1
-                return
-        self.generic_visit(node)
-
-
-def filter_non_mutated(kernel, tensors):
+            nonlocal next_fake_intermediate
+            next_fake_intermediate -= 1
+            return Intermediate(next_fake_intermediate)
+
+        assert isinstance(token, lark.lexer.Token)
+
+        if token.type == "INTERMEDIATE":
+            return Intermediate(int(token.value[len("%") :]))
+        if token.type == "PARAM":
+            return Param(int(token.value[len("%arg") :]))
+
+        raise AssertionError(f"{type(token.type)} => {token.value} invalid")
+
+    # In alternative representation, function names are quoted.
+    # It should be possible to move this into the grammar alltogether.
+    def convert_name(token):
+        s = token.value
+        if s[0] == '"' and s[-1] == '"':
+            return s[1:-1]
+        return s
+
+    @v_args(inline=True)
+    class CalculateOps(Transformer):
+        def process_op_no_ret(self, *args):
+            self.process_op(None, *args)
+
+        def process_op(self, ret, name, args, *rest):
+            ops[convert(ret)] = Op(convert_name(name), convert(args))
+
+    parser = Lark(grammar, parser="lalr", transformer=CalculateOps())
+    parser.parse(ttir)
+    return ops
+
+
+# Given a triton kernel and the arguments for this kernel, this function
+# retrieves the TTIR converted version of the kernel from Triton's API.
+# Parsing the TTIR, it creates a control flow graph, detects all sinks
+# from a predefined list of sinks. NOTE: What if triton exposed this MemWrite Trait
+# From each sink, it traverses the CFG backwards to identify all the input
+# pointers that are mutated
+def identify_mutated_tensors(kernel, kwargs):
+    from triton.compiler.code_generator import ast_to_ttir
     from triton.runtime.autotuner import Autotuner
+    from triton.runtime.jit import JITFunction
+
+    import torch
+    from torch._subclasses.fake_tensor import FakeTensor
 
     if isinstance(kernel, Autotuner):
+        if len(kernel.configs) > 0:
+            # If we are autotuning, then it doesn't matter which version gets
+            # picked for tracing purposes, so lets pick the first one
+            kwargs = {**kwargs, **kernel.configs[0].kwargs}
```

Comment:
If it doesn't the autotuner will will crash since it will be missing arguments?

Reply:
I added a check below

Reply:
> If it doesn't the autotuner will will crash since it will be missing arguments?

I mean, in principle, Triton kernels can have default argument values for non-Tensors? And then you can skip some of those in the config spec. Although, not sure.